### PR TITLE
Correct the number of expected pods created in e2e test case of "retry"

### PR DESCRIPTION
**Pull request by vincent-pli**
# Changes
Fix the test case mentioned in #1976 
This PR only fix the test case not the bug mentioned in the issue, that bug need further research.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

Double check this list of stuff that's easy to miss:

- If you are adding [a new binary/image to the `cmd` dir](../cmd), please update
  [the release Task](../tekton/publish.yaml) to build and release this image.

## Reviewer Notes

If [API changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md) are included, [additive changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#additive-changes) must be approved by at least two [OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS) and [backwards incompatible changes](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-incompatible-changes) must be approved by [more than 50% of the OWNERS](https://github.com/tektoncd/pipeline/blob/master/OWNERS), and they must first be added [in a backwards compatible way](https://github.com/tektoncd/pipeline/blob/master/api_compatibility_policy.md#backwards-compatible-changes-first).

# Release Notes

```
Describe any user facing changes here, or delete this block.

Examples of user facing changes:
- API changes
- Bug fixes
- Any changes in behavior

```

### CODE REVIEW

File: test/retry_test.go [hunk 1]
```diff
@@ -107,9 +108,9 @@ func TestTaskRunRetry(t *testing.T) {
 	pods, err := c.KubeClient.Kube.CoreV1().Pods(namespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("Failed to list Pods: %v", err)
-	} else if len(pods.Items) != numRetries {
+	} else if len(pods.Items) != numRetries+1 {
 		// TODO: Make this an error.
-		t.Logf("BUG: Found %d Pods, want %d", len(pods.Items), numRetries)
+		t.Logf("BUG: Found %d Pods, want %d", len(pods.Items), numRetries+1)
```
Comment (sbwsg) [inline test/retry_test.go]: Is it possible to make this an error now and fail the test?

Comment (vincent-pli) [reply test/retry_test.go]: Sorry, my bad. the PR is only for correcting the test case, the issue mentioned in the issue #1976 is still existed need further research.

I prefer to keep current behavior(not an error just log) until the real issue fixed.

Comment (sbwsg) [reply test/retry_test.go]: Ah I totally understand, thanks @vincent-pli !  Could you update the commit message as well to reflect this?  Something like `Correct the number of expected pods created during retries e2e test`

File: test/retry_test.go [hunk 2]
```diff
@@ -107,9 +108,9 @@ func TestTaskRunRetry(t *testing.T) {
 	pods, err := c.KubeClient.Kube.CoreV1().Pods(namespace).List(metav1.ListOptions{})
 	if err != nil {
 		t.Fatalf("Failed to list Pods: %v", err)
-	} else if len(pods.Items) != numRetries {
+	} else if len(pods.Items) != numRetries+1 {
```
Comment (ImJasonH) [inline test/retry_test.go]: If this is going to be referenced twice, it should probably have a variable, and maybe even a comment. Something like (please feel free to rephrase):

```
// We expect N+1 Pods total, one for each failed and retried attempt, and one for the final attempt.
wantPods := numRetries + 1
```

Comment (vincent-pli) [reply test/retry_test.go]: Done, thanks

**Comment by sbwsg** [comment]
Just pulling a previous comment out of the resolved review comment:

Ah I totally understand, thanks @vincent-pli ! Could you update the commit message as well to reflect this? Something like Correct the number of expected pods created during retries e2e test

**Comment by vincent-pli** [comment]
@sbwsg 
Done, thanks.

**Comment by sbwsg** [comment]
Sorry for the confusion - I meant the commit message, modified by `git commit --amend`.  This is so that the `git log` history shows the updated information.  Thanks and sorry again for the confusion!

**Comment by sbwsg** [comment]
:pray: thank you!

/lgtm

**Comment by dibyom** [comment]
Looks like you need to run goimports. LGTM otherwise 

> I0205 15:43:39.146] ==== RUNNING GOLANGCI-LINT ====
> I0205 15:43:39.146] ===============================
> I0205 15:47:00.972] cmd/creds-init/main.go:19: File is not `goimports`-ed (goimports)
> I0205 15:47:00.992] 	"flag"
> I0205 15:47:00.995] pkg/entrypoint/entrypointer.go:21: File is not `goimports`-ed (goimports)
> I0205 15:47:00.996] 	"go.uber.org/zap"
> I0205 15:47:00.996] cmd/git-init/main.go:19: File is not `goimports`-ed (goimports)
> I0205 15:47:00.996] 	"flag"

**Pull request merged**